### PR TITLE
Reproducible local environment for development and try out

### PR DESCRIPTION
**Pull request by ifraixedes**
The commit messages should contain whatever other information that you'd like to know.

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -41,7 +44,40 @@ Perform a restore:
...
+### Required tools
+
+The local environment requires the following tools:
+
+* [Kind](https://kind.sigs.k8s.io/): Local Kubernetes cluster using Docker containers. It requires to have [Docker](https://www.docker.com/products/docker-desktop) installed.
+* [Kubectl](https://kubernetes.io/docs/tasks/tools/install-kubectl/): The Kubernetes command-line tool.
+* [Velero CLI](https://velero.io/docs/v1.4/basic-install/): The Velero command-line tool.
+
+Be aware that the `Makefile` uses some Unix like tools that you may need to install depending of your Linux distribution and OSX and you will have to install for sure in Windows. While this local environment has been working find in some Linux distributions and probably works in almost all of them, we believe that also works in OSX but we don't know if it will in Windows.
+
```
Comment (Fadila82) [inline README.md:63]: find -> fine

File: Makefile [hunk 2]
```diff
@@ -0,0 +1,94 @@
...
+
+.PHONY: help
+help: ## show this help message
+	@echo $(HELP_MSG)
+	@cat Makefile | awk -F ":.*##"  '/##/ { printf "    %-30s %s\n", $$1, $$2 }' | grep -v  grep
+
+.PHONY: k8s-start
+k8s-start: ## start and initialize a local K8s cluster
+	@if [ ! -d $(K8S_DIR) ]; then \
+		mkdir -p $(K8S_DIR); \
```
[the remark this answers is not in the document]
Comment (ifraixedes) [reply Makefile:30]: Done.
I also added 3 lines to the top that check if the tool binaries are available in the PATH

**Comment by ifraixedes** [comment]
@Fadila82 I made the fixes but squashed the new commits to the corresponding commits.

File: Makefile [hunk 3]
```diff
@@ -0,0 +1,99 @@
...
+.PHONY: velero-uninstall
+velero-uninstall: ## uninstall Velero from the K8s cluster
+ifeq ($(call is_k8s_running),)
+	@kubectl delete namespace/velero clusterrolebinding/velero
+	@kubectl delete crds -l component=velero
+endif
+
+.PHONY: plugin-build
+plugin-build: ## build docker image of this Velero plugin for development
+	@docker image build --squash --rm -t $(DOCKER_IMAGE):$(DOCKER_TAG) .
```
Comment (Fadila82) [inline Makefile:69]: `--squash` seems to be supported only in docker experimental mode

File: Makefile [hunk 4]
```diff
@@ -0,0 +1,99 @@
...
+STORJ_ACCESS ?= # Required
+DOCKER_IMAGE := storjthirdparty/velero-plugin
+DOCKER_TAG ?= dev
+VELERO_BACKUP_LOCATION := default
+
+export KUBECONFIG
+
+# check minimun required tools
+$(if $(shell which kind),,$(error kind tool is required))
+$(if $(shell which velero),,$(error velero tool is required))
```
Comment (Fadila82) [inline Makefile:14]: These checks do not work on my side. `kind` is in my path, but it does not seem to find it (everything works fine if I comment those lines).
I'm not familiar with this syntax, but if I do something like:
`$(if $(shell which kubectl),$(error it is fine)),$(error kubectl tool is required))`
I do get the error "it is fine".

File: Makefile (hunk 3 above)
Comment (ifraixedes) [reply Makefile:69]: oh, I didn't know, I will remove it.

File: Makefile (hunk 4 above)
Comment (ifraixedes) [reply Makefile:14]: :thinking: probably it's due to some differences between `make` versions.
I will remove them.

**Pull request merged**